test/other/embind_tsgen.d.ts

```diff
@@ -88,6 +88,7 @@ export interface MainModule {
   a_bool: boolean;
   an_int: number;
   global_fn(_0: number, _1: number): number;
+  optional_test(_0: Foo | undefined): number | undefined;
```

Comment:
It could probably be interesting to add logic to detect `std::optional<...>` at the end and transform them into `_0?: Foo` instead, as in, actual optional params. Can be a separate PR though.